src/js/tabs/pid_tuning.js

```diff
@@ -2192,3 +2298,177 @@ TABS.pid_tuning.updatePIDColors = function(clear = false) {
     setTuningElementColor($('.pid_tuning .PITCH input[name="f"]'), ADVANCED_TUNING_ACTIVE.feedforwardPitch, ADVANCED_TUNING.feedforwardPitch);
     setTuningElementColor($('.pid_tuning .YAW input[name="f"]'), ADVANCED_TUNING_ACTIVE.feedforwardYaw, ADVANCED_TUNING.feedforwardYaw);
 };
+
+TABS.pid_tuning.changeRatesType = function(rateTypeID) {
+    var self = this;
+    var dialogRatesType = $('.dialogRatesType')[0];
+    var sameRatesType = true;
+    
+    self.currentRatesType = rateTypeID;
+            
+    if (self.currentRatesType != RC_tuning.rates_type) {
+        sameRatesType = false;
+        dialogRatesType.showModal();
+
+        $('.dialogRatesType-cancelbtn').click(function() {
+            sameRatesType = true;
+            self.currentRatesType = RC_tuning.rates_type;
+            $('.rates_type select[id="ratesType"]').val(RC_tuning.rates_type);
+            changeRatesSystem();
+            dialogRatesType.close();
+        });
+
+        $('.dialogRatesType-confirmbtn').click(function() {
+            changeRatesSystem();
+            dialogRatesType.close();
+        });
+    } else {
+        changeRatesSystem();
+        return;
+    }
+    
+    function changeRatesSystem() {
+        self.changeRatesTypeLogo();
+        
+        var rc_rate_max = 2.55, rc_rate_min = 0.01, rc_rate_step = 0.01;
+        var rate_max = 1.0, rate_min = 0, rate_step = 0.01;
+        var expo_max = 1.0, expo_min = 0, expo_step = 0.01;
+        
+        var rc_rate_default = (1).toFixed(2), rate_default = (0.7).toFixed(2), expo_default = (0).toFixed(2); // default values for betaflight curve. all the default values produce the same betaflight default curve (or at least near enough)
+        
+        if (sameRatesType) { // if selected rates type is different from the saved one, set values to default instead of reading
+            $('.pid_tuning input[name="pitch_rate"]').val(RC_tuning.pitch_rate.toFixed(2));
+            $('.pid_tuning input[name="roll_rate"]').val(RC_tuning.roll_rate.toFixed(2));
+            $('.pid_tuning input[name="yaw_rate"]').val(RC_tuning.yaw_rate.toFixed(2));
+            $('.pid_tuning input[name="rc_rate_pitch"]').val(RC_tuning.rcPitchRate.toFixed(2));
+            $('.pid_tuning input[name="rc_rate"]').val(RC_tuning.RC_RATE.toFixed(2));
+            $('.pid_tuning input[name="rc_rate_yaw"]').val(RC_tuning.rcYawRate.toFixed(2));
+            $('.pid_tuning input[name="rc_pitch_expo"]').val(RC_tuning.RC_PITCH_EXPO.toFixed(2));
+            $('.pid_tuning input[name="rc_expo"]').val(RC_tuning.RC_EXPO.toFixed(2));
+            $('.pid_tuning input[name="rc_yaw_expo"]').val(RC_tuning.RC_YAW_EXPO.toFixed(2));
+        }
+
+        switch(self.currentRatesType) {
+            case 1: // RaceFlight
```

Comment:
Instead of using magic numbers here and in other places, and then requiring comments, it would be more elegant, and easier to maintain if you defined constants for them and then used them everywhere:

```
const RATES_TYPES: {
  BETAFLIGHT: 0,
[...]
}
```

and then

```
case RATES_TYPES.RACEFLIGHT:
```